test/widgets/inbox_test.dart

```diff
@@ -595,6 +607,38 @@ void main() {
           check(rectAfterTap).equals(rectBeforeTap);
         });
 
+        testWidgets('shows archived label for archived streams', (tester) async {
+          final stream = eg.stream(streamId: 1, isArchived: true);
+          final subscription = eg.subscription(stream);
+          await setupPage(tester,
+            streams: [stream],
+            subscriptions: [subscription],
+            unreadMessages: [eg.streamMessage(stream: stream)]);
+          await tester.pumpAndSettle();
+
+          final headerRowFinder = findStreamHeaderRow(tester, stream.streamId);
+          check(headerRowFinder).isNotNull();
+
+          final richTextFinder = find.descendant(
+            of: find.byWidget(headerRowFinder!),
+            matching: find.byWidgetPredicate((widget) =>
+              widget is RichText &&
+              widget.text.toPlainText().contains(stream.name)
+            ),
+          );
+          expect(richTextFinder, findsOneWidget);
```

Comment:
Looking at the implementation of `findStreamHeaderRow`, this stanza seems to be repetitive. Let's just remove this check. Another benefit of removing this is that we can focus more on checking the presence of the "(archived)" label.